l10n_ch_lsv_dd/wizard/dd_export_wizard.py

```diff
@@ -0,0 +1,497 @@
+
+import base64
+import collections
+from datetime import date, datetime, timedelta
+from openerp import models, fields, api, _, netsvc, exceptions
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+from openerp.tools import mod10r
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class post_dd_export_wizard(models.TransientModel):
+
+    ''' Postfinance Direct Debit file generation wizard.
+        This wizard is called when the "make payment" button on a
+        direct debit order with payment type "Postfinance DD" is pressed.
+    '''
+    _name = 'post.dd.export.wizard'
+    _description = 'Export Postfinance Direct Debit File'
+
+    currency = fields.Selection(
+        [('CHF', 'CHF'), ('EUR', 'EUR')], _('Currency'),
+        required=True,
+        default='CHF'
+    )
+    banking_export_ch_dd_id = fields.Many2one(
+        'banking.export.ch.dd',
+        _('Direct Debit file'),
+        readonly=True
+    )
+    file = fields.Binary(
+        string=_('File'),
+        related='banking_export_ch_dd_id.file'
+    )
+    filename = fields.Char(
+        string=_('Filename'),
+        related='banking_export_ch_dd_id.filename',
+        size=256,
+        readonly=True
+    )
+    nb_transactions = fields.Integer(
+        string=_('Number of Transactions'),
+        related='banking_export_ch_dd_id.nb_transactions',
+        readonly=True
+    )
+    total_amount = fields.Float(
+        string=_('Total Amount'),
+        related='banking_export_ch_dd_id.total_amount',
+        readonly=True
+    )
+    state = fields.Selection(
+        [('create', _('Create')), ('finish', _('Finish'))],
+        _('State'),
+        readonly=True,
+        default='create'
+    )
+
+    @api.multi
+    def generate_dd_file(self):
+        ''' Generate direct debit export object including the direct
+            debit file content.
+            Called by generate button
+        '''
+        self.ensure_one()
+        payment_order_obj = self.env['payment.order']
+        payment_line_obj = self.env['payment.line']
+
+        active_ids = self.env.context.get('active_ids', [])
+        if not active_ids:
+            raise exceptions.ValidationError(_('No payment order selected'))
+
+        payment_order_ids = payment_order_obj.browse(active_ids)
+        properties = self._setup_properties(payment_order_ids[0])
+        records = []
+        overall_amount = 0
+
+        for payment_order in payment_order_ids:
+            overall_amount += payment_order.total
+            # Order payment_lines to simplify the setup of 'group orders'
+            order_by = ''
+            if payment_order.date_prefered == 'due':
+                order_by = 'account_move_line.date_maturity ASC, '
+            order_by += 'payment_line.bank_id'
+
+            # A direct db query is used because order parameter in
+            # model.search doesn't work over function fields

```

Comment:
But function fields if not stored won't be in database right?

Order by stored function field should work